File: examples/ruby/blink.rb
require 'pi_piper'

pin_out = 17
pin = PiPiper::Pin.new(:pin => pin_out, :direction => :out)

def blink(pin)
	pin.on
	sleep(0.5)
	pin.off
	sleep(0.5)
end

5.times { blink(pin) }

File: examples/python/blink.py
import RPi.GPIO as GPIO
import time

pinOut = 17

GPIO.setmode(GPIO.BCM)
GPIO.setwarnings(False)
GPIO.setup(pinOut, GPIO.OUT)

def blink(pin):
    GPIO.output(pin,GPIO.HIGH)
    time.sleep(0.5)
    GPIO.output(pin,GPIO.LOW)
    time.sleep(0.5)
    return

for i in range(0,5):
    blink(pinOut)

GPIO.cleanup()